src/payment/checkout/payment-form/flex-microform/FlexMicroform.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { setPaymentTokenFetch } from '../../../payment-methods/cybersource/service';
+
+class FlexMicroform extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            microform: null,
+        }
+        this.onLoad = this.onLoad.bind(this);
+        this.getPaymentToken = this.getPaymentToken.bind(this);
+    }
+
+    componentDidMount() {
+        setPaymentTokenFetch(this.getPaymentToken);
+
+        const script = document.createElement('script');
+        script.src = "https://flex.cybersource.com/cybersource/assets/microform/0.11/flex-microform.min.js";
+        script.async = true;
+
+        script.addEventListener('load', this.onLoad);
+        script.addEventListener('error', e => {
+            console.error(e);
+        });
+
+        document.body.appendChild(script);
+    }
+
+    initialize() {
+        if (this.state.microform !== null) {
+            return;
+        }
+        if (!this.props.captureKeyId) {
+            return;
+        }
+        if (window['Flex'] === undefined) {
+            return;
+        }
+        const microform = new window.Flex(this.props.captureKeyId).microform({
+            styles: { 
+                'input': {    
+                    'font-size': '16px',
+                    'font-family': 'Roboto, "Helvetica Neue", Arial, sans-serif',
+                    'color': '#2d323e',
+                },
+                ':disabled': {
+                    'cursor': 'not-allowed'
+                }
+            }
+        });
+        this.setState({microform: microform});
+    }
+
+    getPaymentToken(cardExpirationMonth, cardExpirationYear) {
```

Comment:
Okay, because you want to call this function from your cybersource/service.js file, I'd probably rearrange the code to get make it possible to co-locate this function with that code.  The part where you have to save a reference to this in service.js by exporting the setter is a bit odd.

For instance, I'd favor stashing a reference to `microform` _globally_, moving this function to service.js, and hitting up that global reference for `createToken` below, rather than needing to jump through these hoops. That wouldn't be great either, but it might be clearer what's going on.

Another option - depending on how long-lived the token created by `createToken` below is - would be to have this component generate the token on component mount/microform initializatin and have it passed through (along with all the other data) to the `checkoutWithToken` function - that way it doesn't have to "reach out" at all/can just use its parameters.

Reply:
I really like the idea of putting the `microform` instance in global scope, that's probably the simplest thing to do.

I suppose we could just pass around a state variable through the component tree to tell the credit card number field and the CVV code field to re-render when the global instance had been initialized.

Reply:
Note that we have to wait until the user has filled in their credit card number and clicked "Place Order" prior to calling `createToken` so I don't think we can do this on mount/initialization.